code/Controllers/SilverStripeNavigatorItem_ArchiveLink.php

```diff
@@ -9,6 +9,10 @@
 use SilverStripe\ORM\FieldType\DBField;
 use SilverStripe\Versioned\Versioned;
 
+/**
+ * Class have been moved from `silverstripe/cms` to `silverstripe/admin` and renamed.
+ * @deprecated Use SilverStripe\VersionedAdmin\Navigator\SilverStripeNavigatorItem_ArchiveLink instead
+ */
 class SilverStripeNavigatorItem_ArchiveLink extends SilverStripeNavigatorItem
```

Comment:
Add a new constructor for this with the appropriate deprecation notice
```php
public function __construct(CMSPreviewable $record)
{
    /* deprecation notice here */
    parent::__construct($record);
}
```
Same with all the other `SilverStripeNavigatorItem` subclasses.

Reply:
DONE